examples/alternate-receive/alternate_receive.bal

```diff
@@ -0,0 +1,48 @@
+import ballerina/http;
+import ballerina/io;
+import ballerina/lang.runtime;
+
+public function main() {
+    // Both arguments passed to the function are valid URLs.
+    // Thus alternate receive in worker `w3` sets the
+    // first value it rececives from a worker as the result.
+    fetchFirst("https://postman-echo.com/get?worker=w1", "https://postman-echo.com/get?worker=w2");
+
+    // The first argument passed to the function is an invalid URL.
+    // The worker `w1` in `fetchFirst` function returns an error.
+    // Thus alternate receive in worker `w3` waits further and sets
+    // the value that is received from `w2` as the result.
+    fetchFirst("https://postman-echo.com/ge?worker=w4", "https://postman-echo.com/get?worker=w5");
+}
+
+function fetchFirst(string url1, string url2) {
+    worker w1 {
+        map<json>|error result = fetch(url1);
+        result -> w3;
+    }
+
+    worker w2 {
+        runtime:sleep(3);
+        map<json>|error result = fetch(url2);
+        result -> w3;
+    }
+
+    worker w3 returns json|error? {
+        // The value of the variable `result` is set as soon as
+        // a non-error message is received from either worker `w1` or `w2`.
+        map<json>|error result = <- w1 | w2;
+        return getJsonProperty(result);
+    }
+
+    json|error? w3Result = wait w3;
+    io:println(w3Result);
+}
+
+function fetch(string url) returns map<json>|error {
+    http:Client cl = check new (url);
+    record {map<json> args;} payload = check cl->get("");
+    return payload.args;
```

Comment:
Shall we introduce a record and bind to that instead? Can change getJsonProperty also then.